apps/system/test/unit/bluetooth_v2_test.js

```diff
@@ -0,0 +1,328 @@
+
+    test('return true when profile is connected', function() {
+      profiles.forEach(function(profile) {
+        Bluetooth['_' + profile + 'Connected'] = false;
+      });
+      profiles.forEach(function(profile) {
+        assert.isFalse(Bluetooth.isProfileConnected(profile));
+      });
+    });
+
+    test('return true when profile is connected', function() {
+      profiles.forEach(function(profile) {
+        Bluetooth['_' + profile + 'Connected'] = true;
+      });
+      profiles.forEach(function(profile) {
+        assert.ok(Bluetooth.isProfileConnected(profile));
+      });
+    });
+  });
+
+  suite('updateProfileStat', function() {
+    setup(function() {
+      this.sinon.stub(MockBTAdapter, 'addEventListener');
+      Bluetooth._updateProfileStat(MockBTAdapter);
+    });
+
+    test('return true when profile is connected', function() {
+      assert.ok(MockBTAdapter.addEventListener.calledWith('hfpstatuschanged'));
+      assert.ok(MockBTAdapter.addEventListener.calledWith('a2dpstatuschanged'));
+      assert.ok(MockBTAdapter.addEventListener.calledWith('scostatuschanged'));
+    });
+  });
+
+  suite('Initialize', function() {
+    setup(function() {
+      this.sinon.spy(navigator.mozBluetooth, 'addEventListener');
+      this.sinon.stub(Bluetooth, '_setProfileConnected');
+      this.sinon.spy(window, 'addEventListener');
+      this.sinon.spy(window, 'dispatchEvent');
+      this.sinon.stub(Service, 'registerState');
+      Bluetooth.start();
+    });
+
+    test('defaultAdapter is called', function() {
+      assert.equal(Bluetooth._adapter, navigator.mozBluetooth.defaultAdapter);
+    });
+
+    test('listener called', function() {
+      assert.equal(MockSettingsListener.mName, 'bluetooth.enabled');
+      assert.ok(window.navigator.mozBluetooth
+        .addEventListener.calledWith('adapterremoved'));
+      assert.ok(window.addEventListener
+        .calledWith('request-enable-bluetooth'));
+      assert.ok(window.addEventListener
+        .calledWith('request-disable-bluetooth'));
+    });
+
+    test('MessageHandler bluetooth-opp-transfer-start is called',
+      function() {
+        MockNavigatormozSetMessageHandler.mTrigger(
+          'bluetooth-opp-transfer-start', {
+            source: {
+              data: {}
+            }
+        });
+
+        assert.ok(Bluetooth._setProfileConnected.called);
+        assert.ok(window.dispatchEvent.called);
+    });
+
+    test('MessageHandler bluetooth-opp-transfer-complete is called',
+      function() {
+        MockNavigatormozSetMessageHandler.mTrigger(
+          'bluetooth-opp-transfer-complete', {
+            source: {
+              data: {}
+            }
+        });
+
+        assert.ok(Bluetooth._setProfileConnected.called);
+        assert.ok(window.dispatchEvent.called);
+    });
+
+    test('register state', function() {
+      assert.ok(Service.registerState.calledWith('isEnabled'));
+    });
+
+    // test('Should lazy load icons', function() {
```

Comment:
this part is preserved here since the statusbar icon patch is about to land. Could remove these if statusbar icon patch is not landed before this patch